src/logic/safe/transactions/gas.ts

```diff
@@ -92,6 +93,141 @@ export const estimateTxGasCosts = async (
+      // In that case we check if there is empty data (which means an error)
+      // Or if there is a valid estimation amount
+      const estimationData = getNonGethErrorDataResult(error.message)
+
+      if (!estimationData || sameString(estimationData, EMPTY_DATA)) {
+        throw error
+      }
+
+      return {
+        gasEstimationResponse: estimationData,
+        isGethNode: false,
+      }
+    })
+}
+
+const calculateGasForGethNodes = async (
+  additionalGasBatches: number[],
+  safeAddress: string,
+  estimateData: string,
+  txGasEstimation: number,
+  dataGasEstimation: number,
+): Promise<number> => {
+  const web3 = getWeb3()
+  const batch = new web3.BatchRequest()
+  const estimationRequests = additionalGasBatches.map(
+    (additionalGas) =>
+      new Promise((resolve) => {
+        // there are no type definitions for .request, so for now ts-ignore is there
+        // Issue link: https://github.com/ethereum/web3.js/issues/3144
+        // eslint-disable-next-line
+        // @ts-ignore
+        const request = web3.eth.call.request(
+          {
+            to: safeAddress,
+            from: safeAddress,
+            data: estimateData,
+            gasPrice: 0,
+            gasLimit: txGasEstimation + dataGasEstimation + additionalGas,
+          },
+          (error, res) => {
+            // res.data check is for OpenEthereum/Parity revert messages format
+            const isOpenEthereumRevertMsg = res && typeof res.data === 'string'
+            const isEstimationSuccessful =
+              !error &&
+              ((typeof res === 'string' && res !== EMPTY_DATA) ||
+                (isOpenEthereumRevertMsg && res.data.slice(9) !== EMPTY_DATA))
+
+            resolve({
+              success: isEstimationSuccessful,
+              estimation: txGasEstimation + additionalGas,
+            })
+          },
+        )
+
+        batch.add(request)
+      }),
+  )
+  batch.execute()
+
+  const estimationResponses = await Promise.all(estimationRequests)
+  const firstSuccessfulRequest: any = estimationResponses.find((res: any) => res.success)
+
+  if (firstSuccessfulRequest) {
+    return firstSuccessfulRequest.estimation
+  }
+  return 0
+}
+
+const getNonGethErrorDataResult = (errorMessage: string): string | undefined => {
+  // Extracts JSON object from the error message
+  const [, ...error] = errorMessage.split('\n')
+  const errorAsJSON = JSON.parse(error.join(''))
+
+  if (errorAsJSON?.data) {
+    const [, dataResult] = errorAsJSON.data.split('Reverted ')
+    return dataResult
+  }
+}
+
+const calculateGasForNonGethNodes = async (
+  additionalGasBatches: number[],
+  safeAddress: string,
+  estimateData: string,
+  txGasEstimation: number,
+  dataGasEstimation: number,
+): Promise<number> => {
+  const web3 = getWeb3()
+  for (const additionalGasIterator of additionalGasBatches) {
+    try {
+      await web3.eth.call({
```

Comment:
Why did we remove the batch request support?

Reply:
Because for batch requests we need to use `web3.eth.call.request()` instead of `web3.eth.call()`. The first method does not returns the error as data that we need.

Responses examples:

Batch request:
`Error: Returned error: Internal JSON-RPC error.`

Non-batch request:

```
 Error: Internal JSON-RPC error.
{
  "code": -32015,
  "message": "VM execution error.",
  "data": "Reverted 0x08c379a0000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000006457"
}
```

Reply:
:thinking: hmmm maybe in this case we should adjust it for all ... while we might have to do 1 or 2 more request the code overhead might not be worth it.

Reply:
what do you think? @fernandomg

Reply:
@rmeissner, by that you mean to get rid of the `batchRequest` approach?

I'm perfectly with that.

It's easy to follow, in the worst-case scenario we'll do 10 RPC requests (one after the other, not triggered in parallel), and the user may experiment a delay (we should think of adding something around "calculating gas..." statement somewhere, but in a different issue).

Reply:
yes ... but maybe @dasanra or @mikheevm have a better idea ... but since batched requests truncate the response it is not really possible to always use it.

Reply:
> we'll do 10 RPC requests (one after the other, not triggered in parallel)
that's because we are using await, right? if we make use of `promise.all()` it should not be the case, isn't it? 
@fernandomg

Reply:
I think not doing them in parallel might be better here, as in most cases we will probably only need 1 or 2 requests, perform all 10 for this always would be quite some additional overhead.

Reply:
I removed the batch request and simplified the logic, now it doesn't matter which node it's running the code, the function `getGasEstimationTxResponse` will take care of it and return a result, then I replaced `calculateGasForOpenEthereumNodes` / `calculateGasForGethAndNethermindNodes` with the function `calculateMinimumGasForTransaction`. I think it's more clear now, please give it a check